packages/devextreme/js/ui/text_box/ui.text_editor.mask.js

```diff
@@ -581,8 +572,24 @@ const TextEditorMask = TextEditorBase.inherit({
             default:
                 this.callBase(args);
         }
-    }
+    },
+
+    _dispose: function() {
+        clearTimeout(this._caretTimeout);
+        this.callBase();
+    },
 
+    clear: function() {
+        if(this._maskRulesChain) {
+            this._maskRulesChain.clear(this._normalizeChainArguments());
+            this._updateMaskInfo();
+            const caret = this._maskRulesChain.first();
+            this._caretTimeout = setTimeout(() => {
```

Comment:
We should not use setTimeout here. Such a code always makes future maintenance much more difficult.
How now your solution works:
1.`_clear` method clears maskChain
2. In callbase we call `renderValue` which updates `text` by maskChain's text (empty now)
3. TextBox renders input value in a usual way by setting `.val` and it moves caret to the end of the string
4. Your setTimeout sets caret after the first character.

We can get rid of the setTimeout if use more direct (specific for mask) way to render a value - `_renderMaskedValue`.

I suggest you check `_clear` method of a base editor one more time. It basically
- changes value to a default one
- if value is already default, it does what we usually do on option change

It means we can just check what we do in `value` option change handler and emulate this behavior in `_clear` method.

Smth like:
```javascript
clear: function() {
      const { value: defaultValue } = this._getDefaultOptions();
      if(this.option('value') === defaultValue) {
          this._renderMaskedValue();
      }
      this.callBase();
  }
```
Please, check it out and test one more time. It's just an idea, can be not a final solution.